lib/sinon/spy.js

```diff
@@ -152,6 +152,12 @@ function createProxy(func, proxyLength) {
             return p.invoke(func, this, slice(arguments));
         };
     }
+    var nameDescriptor = Object.getOwnPropertyDescriptor(func, "name");
```

Comment:
The code in this change looks very similar to the code in the next change. Would it be wise to extract it to a shared function? Perhaps even put that function in `@sinonjs/commons`?

Reply:
Yes, I had the same thought, but in one case we're just using one descriptor as is, in the other case we modfy the `value`. Extracting a function produces more code and would include a superfluous `value` assignment for one of the cases. So I decided against it for clarity.

I am treating `commons` as a toolbelt for shared functions. This is a Sinon only feature so far. We can move it if we want to reuse it elsewhere.